src/main_menu.cpp

```diff
@@ -611,7 +612,21 @@ bool main_menu::new_character_tab()
                         g->u = player();
                         continue;
                     }
-                    if( !g->u.create( sel2 == 0 ? PLTYPE_CUSTOM : ( sel2 == 2 ? PLTYPE_RANDOM : PLTYPE_NOW ) ) ) {
+                    character_type play_type = PLTYPE_NULL;
```

Comment:
Have you added the `PLTYPE_NULL` enumeration value only to use it here for initialization that will be overridden anyway? You can initialize this variable with whatever value you like, a new value is assigned directly afterwards.

Reply:
I tried NULL and nullptr, but compiler throw errors on these, so I added this enum.

Reply:
> I tried NULL and nullptr, but compiler throw errors on these,

Well, because `NULL` and `nullptr` are for pointers, but this variable is not a pointer at all, so they won't work.

> so I added this enum.

You could have used any other existing enumeration value, as I said above, it is overridden anyway.